plonky2/src/fri/batch_verifier.rs

```diff
@@ -220,18 +220,16 @@ fn batch_fri_verifier_query_round<
         if batch_index < degree_logs.len()
             && n == degree_logs[batch_index] + params.config.rate_bits
         {
-            let subgroup_x_init = F::MULTIPLICATIVE_GROUP_GENERATOR
-                * F::primitive_root_of_unity(n).exp_u64(reverse_bits(x_index, n) as u64);
             let eval = batch_fri_combine_initial::<F, C, D>(
                 instance,
                 batch_index,
                 &round_proof.initial_trees_proof,
                 challenges.fri_alpha,
-                subgroup_x_init,
+                subgroup_x,
```

Comment:
This `subgroup_x` works now, since the polynomials in batch are evaluated over same domain.